scalameta/contrib/shared/src/main/scala/scala/meta/contrib/instances/ExtractStatSubtypeInstances.scala

```diff
@@ -0,0 +1,53 @@
+package scala.meta.contrib.instances
+
+import scala.meta._
+import scala.meta.contrib._
+
+trait ExtractStatSubtypeInstances {
+  implicit def extractDefsFromStats[A](implicit ev: Extract[A, Stat]): Extract[A, Defn.Def] =
+    Extract(a => ev.extract(a).collect { case d: Defn.Def => d })
```

Comment:
We can probably abstract over this boilerplate with a ClassTag. However, this is fine by me.